### ISSUE TITLE
PropTypes not recognized when class not at top scope

### ISSUE THREAD

**Issue by lemonmade** [yannickcr/eslint-plugin-react#264]
In an example such as this, where a React class is constructed as part of a function call:

```js
import React from 'react-native';

export function createReactClass() {
  class ReactComponent extends React.Component {
    render() {
      let {aProp} = this.props;
      return aProp;
    }
  }

  ReactComponent.propTypes = {aProp: React.PropTypes.string};

  return ReactComponent;
}
```

The `prop-types` rule incorrectly marks `aProp` as being missing from the props validation. If this is changed to use a `static propTypes`, it works as expected (I can't use `static` because Flow doesn't support it, unfortunately).

**Comment by lemonmade** [comment]
Thank you for resolving this so quickly!
